plugins/uefi/fu-self-test.c

```diff
@@ -39,54 +41,36 @@ fu_uefi_pcrs_1_2_func (void)
 static void
 fu_uefi_pcrs_2_0_func (void)
 {
+	g_autofree gchar *devpath = NULL;
+	g_autofree gchar *sysfstpmdir = NULL;
+	const gchar *tpm_server_running = g_getenv ("TPM_SERVER_RUNNING");
 	gboolean ret;
 	g_autoptr(FuUefiPcrs) pcrs = fu_uefi_pcrs_new ();
 	g_autoptr(GError) error = NULL;
 	g_autoptr(GPtrArray) pcr0s = NULL;
-	g_autoptr(GPtrArray) pcr1s = NULL;
 	g_autoptr(GPtrArray) pcrXs = NULL;
 
-	g_setenv ("FWUPD_UEFI_TPM2_YAML_DATA",
-		  "sha1 :\n"
-		  "  0  : cbd9e4112727bc75761001abcb2dddd87a66caf5\n"
-		  "sha256 :\n"
-		  /* old output format of tpm2_listpcrs and tpm2_pcrlist */
-		  "  0  : 122de8b579cce17b0703ca9f9716d6f99125af9569e7303f51ea7f85d317f01e\n"
-		  /* new output format of tpm2_pcrread */
-		  "  1 : 0x0D89E7CA2C487EED36DB5684826B4ABF0904D3BDCD74E61999573570CF3F9C75\n", TRUE);
+	g_unsetenv ("FWUPD_SYSFSTPMDIR");
```

Comment:
Unfortunately yes because the environment variables are not scoped: `FWUPD_SYSFSTPMDIR` is still set from `fu_uefi_pcrs_1_2_func`, so `fu_uefi_pcrs_setup` will find a TPM 1.2 device if the variable is not unset first. I just moved the environment variable definition from `fu_uefi_ucs2_func` to `fu_uefi_pcrs_1_2_func` so that the order in which `fu_uefi_pcrs_1_2_func` and `fu_uefi_pcrs_2_0_func` are run does not matter.

Reply:
or unset it in the TPM 1.2 function?

Reply:
Ah yes, that is cleaner indeed. I was worried first that `g_unsetenv` might be skipped if an assertion in `fu_uefi_pcrs_1_2_func` fails, but in that case complete test program is aborted, so this approach should be safe.